pandas/core/generic.py

```diff
@@ -7995,24 +7995,29 @@ def _clip_with_scalar(self, lower, upper, inplace: bool_t = False):
         ):
             raise ValueError("Cannot use an NA value as a clip threshold")
 
-        result = self
-        mask = isna(self._values)
+        mgr = self._mgr
 
-        with np.errstate(all="ignore"):
+        if inplace:
             if upper is not None:
-                subset = self <= upper
-                result = result.where(subset, upper, axis=None, inplace=False)
+                cond = self > upper
+                mgr = mgr.putmask(mask=cond, new=upper, align=False)
             if lower is not None:
-                subset = self >= lower
-                result = result.where(subset, lower, axis=None, inplace=False)
-
-        if np.any(mask):
-            result[mask] = np.nan
+                cond = self < lower
+                mgr = mgr.putmask(mask=cond, new=lower, align=False)
+        else:
+            mask = isna(self)
+            if upper is not None:
+                cond = mask | (self <= upper)
+                mgr = mgr.where(other=upper, cond=cond, align=False)
+            if lower is not None:
+                cond = mask | (self >= lower)
```

Comment:
Can we use > or >= in both cases? Also why don’t we need a mask in the inplace case?

Reply:
The condition in `putmask`  identifies the values to be updated so we look for values outside the threshold (e.g. ">")

The condition in `where` identifies the values to be left as is which are NA values (mask) and anything within the threshold (e.g. "<=")

So I think the difference is consistent with the meaning of those different methods. Using <= and >= in putmask might would actually turn a no-ops at the boundary into unnecessary ops I think.

Reply:
Thx, makes sense. Did not consider that, could you add a really short comment to that effect?